### PR TITLE
(#350) CMS Site Section tests 

### LINKED ISSUE #350: Site section : setting site section as nav root
## As a [role], I want [feature], so that [benefit]

> **ESTIMATE** TBD

### Acceptance Criteria

```gherkin
 Scenario: making site section a nav root
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Structure" tab
        And user clicks on "Taxonomy" sub tab
        And user selects "List terms" option from Operations for "Site Sections"
        And user clicks on children link for "Home"
        And user clicks on children link for "About Cancer"
        And user clicks on children link for "Coping with Cancer"
        And user selects "Edit" operation for "Test Site Section"
        Then page title is "Edit term"
        And user checks "Set Section Nav Root" checkbox to set as a nav root
        When user saves the content page


    Scenario: Verify that the test site section points to a new url after changing landing's page site section
        Given user is navigating to the front end site with path "/about-cancer/coping/test-site-section"
        Then left navigation does not display "About Cancer"
        And the current page is "Test Site Section" in left nav
        And page title is "Article to test Site Section"
```

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by alinai**
Closes #350 

Added feature file.

### CODE REVIEW

**Comment by alinai** [comment]
cypress test run on test-acsf:
image

File: cypress/e2e/CMS/$SiteSections.feature [hunk 1]
```diff
@@ -77,7 +101,8 @@ Feature: As a cms user I want to be able to create Site Section to promote Site
         And user selects "children" link under "Home"
         And user selects "children" link under "About Cancer"
         And user selects "children" link under "Coping with Cancer"
-        And user clicks on "Delete" from  dropdown button under "Test Site Section"
+        And user selects "Delete" operation for "Test Site Section"
+        #And user clicks on "Delete" from  dropdown button under "Test Site Section"
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$SiteSections.feature:105]: is this is not needed, then remove this step)

File: cypress/e2e/CMS/$SiteSections/index.js [hunk 2]
```diff
@@ -90,6 +90,21 @@ And('user selects {string} link under {string}', (children, parentName) => {
     cy.get(`a.menu-item__link:contains("${parentName}")`).parent().parent().find(`a:contains("${children}")`).click();
 });
 
-And('user clicks on {string} from  dropdown button under {string}', (deleteBtn, termName) => {
-    cy.get(`li.delete.dropbutton-action a:contains("${deleteBtn}")`).first().click({ force: true });
-});
\ No newline at end of file
+// And('user clicks on {string} from  dropdown button under {string}', (deleteBtn, termName) => {
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$SiteSections/index.js:93]: commented out, should be removed then

File: cypress/e2e/CMS/$SiteSections/index.js [hunk 3]
```diff
@@ -90,6 +90,21 @@ And('user selects {string} link under {string}', (children, parentName) => {
...
+And('user selects {string} operation for {string}', (editBtn, termName) => {
+    cy.get(`ul.dropbutton a:contains("${editBtn}")`).first().click({ force: true });
+});
+
+And('user checks {string} checkbox to set as a nav root', (setNavRoot) => {
+    cy.get('#edit-field-section-nav-root-wrapper > .js-form-item').find('input').check();
+});
+
+Then('left navigation does not display {string}', (leftNavItem) => {
+    cy.get(`nav[aria-label='Secondary navigation']`).find('a').contains(new RegExp(`^${leftNavItem}$`)).should('not.exist');
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$SiteSections/index.js:106]: do you need regex here? Doesn't it work with just a string?

**Review by KateMashkinaNIH — CHANGES_REQUESTED** [review]

**Comment by alinai** [comment]
cypress test run on test-acsf:

image

File: cypress/e2e/CMS/$SiteSections/index.js (hunk 3 above)
Comment (alinai) [reply cypress/e2e/CMS/$SiteSections/index.js]: fixed

File: cypress/e2e/CMS/$SiteSections/index.js (hunk 2 above)
Comment (alinai) [reply cypress/e2e/CMS/$SiteSections/index.js]: Removed

File: cypress/e2e/CMS/$SiteSections.feature (hunk 1 above)
Comment (alinai) [reply cypress/e2e/CMS/$SiteSections.feature]: Removed

**Comment by alinai** [comment]
@KateMashkinaNIH , updated per your review. Thanks!

**Comment by alinai** [comment]
npm test run on dev-acsf:

image

**Pull request merged**